### PR TITLE
feat: add breadcrumbs navigator

### PR DESCRIPTION
**Pull request by rayacers**
Add the breadcrumbs navigator.
Screenshot 2023-03-29 at 5 50 37 PM

### CODE REVIEW

File: src/components/navigator/Navigator.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    pathname !== HOME_PATH
+  ) {
+    return <Navigate to={HOME_PATH} />;
+  }
+
+  return (
+    <Breadcrumbs
+      separator={<NavigateNextIcon />}
+      aria-label="breadcrumb"
+      style={{ backgroundColor: '#f6f6f6' }}
```
Comment (pyphilia) [inline src/components/navigator/Navigator.tsx:114]: Factor out the color in a variable in `constants`.

File: src/components/navigator/util.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import { Link } from 'react-router-dom';
+
+import { Typography, styled } from '@mui/material';
+
+import { ITEM_NAME_MAX_LENGTH } from '../../config/constants';
+import { buildItemPath } from '../../config/paths';
+
+const StyledLink = styled(Link)({
+  textDecoration: 'none',
+  color: '#000000',
```
Comment (pyphilia) [inline src/components/navigator/util.tsx:12]: Why did you need to use black? Was the link blue? Try to use the `theme` color instead. I think `text.primary` might work? https://mui.com/system/palette/

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the PR, I've left a bunch of comments, let me know if you have questions.

File: src/components/navigator/Menu.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+  if (item.size === 0) {
+    return null;
+  }
+  return (
+    <>
+      <StyledIconButton
+        onClick={handleClick}
+        aria-controls={open ? 'menu-root' : undefined}
+        aria-haspopup="true"
```
[the remark this answers is not in the document]
Comment (rayacers) [reply src/components/navigator/Menu.tsx:82]: I think it takes a string. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup.

**Comment by rayacers** [comment]
Thanks for the review! I pushed the fix. Please let me know if there is any problem.

**Review by pyphilia — APPROVED** [review]

**Pull request merged**